src/app/files/fileProvider.js

```diff
@@ -131,24 +133,43 @@ class FileProvider {
     return false
   }
 
+  /**
+   * Removes the folder recursively
+   * @param {*} path is the folder to be removed
+   */
   remove (path) {
-    var unprefixedpath = this.removePrefix(path)
-    if (!this._exists(unprefixedpath)) {
-      return false
-    }
-    const stat = window.remixFileSystem.statSync(unprefixedpath)
-    try {
-      if (stat.isDirectory()) {
-        window.remixFileSystem.rmdirSync(unprefixedpath, console.log)
-      } else {
-        window.remixFileSystem.unlinkSync(unprefixedpath, console.log)
+    path = this.removePrefix(path)
+    if (window.remixFileSystem.existsSync(path)) {
+      const stat = window.remixFileSystem.statSync(path)
+      try {
+        if (!stat.isDirectory()) {
+          window.remixFileSystem.unlinkSync(path, console.log)
+          this.event.trigger('fileRemoved', [this._normalizePath(path)])
+          return true
+        } else {
+          let items = window.remixFileSystem.readdirSync(path)
+          if (items.length !== 0) {
+            items.forEach((item, index) => {
+              let curPath = path + '/' + item
```

Comment:
Maybe : 
```javascript
const currPath = `${path}/${item}`
```

Reply:
why it is better?

Reply:
You don't reassign it so you can you `const`. 
Personally I find the `${a}divder${b}` writing more readable than using concatenation (just suggesting here).